### ISSUE TITLE
colorama is unconditionally imported and initialized which leads to wrapping of stdout/stderr

### ISSUE THREAD

**Issue by yarikoptic** [tqdm/tqdm#178]
ok -- I couldn't let it go, and troubleshooting an awkward bug we have encountered which I could not really figure out (still can't) exactly.  But situation is that in datalad project we provide a git-annex special remote which communicates with git-annex via stdin/stdout.  Mere import of tqdm lead to something in communication not work out as it should, but only on Debian stretch (testing, with colorama 0.3.7), and worked okish on stable and various ubuntus (colorama <= 0.3.2)....  so it might be that difference in behaviour is due to those versions difference but I didn't have time to do all the dance to bisect. Troublshooting further pointed out to `colorama.init`  as the cause of the communication problem.

But is colorama needed only on windows?  
or it is useful to be "initialized" on any system?
how useful it is if interface is not interactive?
I just wondered if use of colorama should be conditioned on the system and in general made optional?

**Comment by lrq3000** [comment]
Yes indeed, colorama is only useful for Windows and should only be loaded
on Windows, so that's weird. Colorama is indeed optional anyway, it only
gets imported if the user has it installed.

We will investigate why colorama is always imported, this should not happen.

2016-06-02 17:12 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

**Comment by lrq3000** [comment]
Why did you assign me? Your PR fixes the issue. Do you want me to help with
something?

2016-06-02 21:01 GMT+02:00 Casper da Costa-Luis <notifications@github.com>:

**Comment by casperdcl** [comment]
ah yes, was wondering if anyone had a neat way to pass through the entire notebook class docstrings without importing it.

**Comment by yarikoptic** [comment]
why I often take upon such challenges! ;)  for some reason this one on numpy doesn't work for me with python3 but I bet it could be easily adopted for tqdm  http://www.onerussian.com/tmp/delayed_adapter.py:
and inline:
```python
class DelayedAdapter(object):
    def __init__(self, mod, name):
        self.__mod = mod
        self.__name = name
        self.__obj = None

    @property
    def _obj(self):
        if self.__obj is None:
            # do the import
            print("Importing %s" % self.__mod)
            #mod = __import__(self.__mod, globals=globals(), locals=locals())
            # or probably more kosher:
            import importlib
            mod = importlib.__import__(self.__mod)
            self.__obj = getattr(mod, self.__name) if self.__name else mod
        return self.__obj

    @property
    def __doc__(self):
        return self._obj.__doc__

    def __call__(self, *args, **kwargs):
        print("Calling")
        return self._obj(*args, **kwargs)

adapter = DelayedAdapter('numpy', 'arange')
print("Asking for __doc__")
print(adapter.__doc__[:10])
print(adapter(3))
```

**Comment by lrq3000** [comment]
Wow nice Yaroslav! Yes that could be worked out to be implemented in tqdm,
I hope it won't be too hard to adapt to work with Python3 (the importlib
changed a lot and lots of interesting features were removed...).

2016-06-03 16:30 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

**Comment by casperdcl** [comment]
closed by #179
